src/modules/iotjs_module_gpio.c

```diff
@@ -198,13 +206,17 @@ jerry_value_t gpio_do_write_or_writesync(const jerry_value_t jfunc,
   } else if (jerry_value_is_boolean(jargv[0])) {
     value = jerry_get_boolean_value(jargv[0]);
   } else {
-    const char* js_create_error_arg;
+    unsigned char err[] = "GPIO Error";
     if (gpio_op == IOTJS_GPIO_WRITE) {
-      js_create_error_arg = iotjs_periph_error_str(kGpioOpWrite);
-    } else {
-      js_create_error_arg = "GPIO WriteSync Error - Wrong argument type";
+      const char* error_msg = iotjs_periph_error_str(kGpioOpWrite);
+      jerry_value_t error_str = jerry_create_string((jerry_char_t*)error_msg);
+      jerry_value_t args[] = { error_str };
+      iotjs_invoke_callback(JS_GET_ARG_IF_EXIST(1, function), jthis, args, 1);
```

Comment:
If there is no callback function, it will create a bug.

For example:
``` js
gpio.write({});
```
You need to change the `function` to execute only when it exists.